modules/nanopolish/1.0/nanopolish.smk

```diff
@@ -0,0 +1,266 @@
+    logger.warning(
+                '\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}'
+                "\n" f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m'
+                )
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section 
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["promethion"]`
+CFG = op.setup_module(
+    name = "nanopolish",
+    version = "1.0",
+    subdirectories = ["inputs", "meth_calls", "meth_freq","merged_nanopolish_calls","merged_nanopolish_freq","outputs"]
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _promethion_input,
+    _install_nanopolish,
+    _promethion_input_chrs,
+    _nanopolish_output,
+    _nanopolish_all
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _promethion_input:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"],
+        fastq = CFG["inputs"]["sample_fastq"],
+        index = CFG["inputs"]["sample_index"],
+        fai = CFG["inputs"]["sample_fai"],
+        gzi = CFG["inputs"]["sample_gzi"],
+        readdb = CFG["inputs"]["sample_readdb"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai",
+        fastq = CFG["dirs"]["inputs"] + "fastq/{seq_type}--{genome_build}/{sample_id}.fastq",
+        index = CFG["dirs"]["inputs"] + "fastq/{seq_type}--{genome_build}/{sample_id}.fastq.index",
+        fai = CFG["dirs"]["inputs"] + "fastq/{seq_type}--{genome_build}/{sample_id}.fastq.index.fai",
+        gzi = CFG["dirs"]["inputs"] + "fastq/{seq_type}--{genome_build}/{sample_id}.fastq.index.gzi",
+        readdb = CFG["dirs"]["inputs"] + "fastq/{seq_type}--{genome_build}/{sample_id}.fastq.index.readdb",
+    run:
+        op.absolute_symlink(input.bam, output.bam)
+        op.absolute_symlink(input.bai, output.bai)
+        op.absolute_symlink(input.fastq, output.fastq)
+        op.absolute_symlink(input.index, output.index)
+        op.absolute_symlink(input.fai, output.fai)
+        op.absolute_symlink(input.gzi, output.gzi)
+        op.absolute_symlink(input.readdb, output.readdb)
+
+# Installs latest nanopolish release from github if the nanopolish folder is not present yet
+rule _install_nanopolish:
+    params:
+        nanopolish = CFG["inputs"]["nanopolish_exec"]
+    output: 
+        complete = CFG["inputs"]["nanopolish_exec"] + "/nanopolish_dependencies_installed.success"
+    shell:
+        '''
+
+        if [ ! -f {params.nanopolish}/nanopolish ]; then
```

Comment:
You shouldn't need the `if` statement since Snakemake will only run a rule if the output file is missing or older than the output of a downstream rule. You can also wrap the output in touch: 
```
complete = touch(CFG["inputs"]["nanopolish_exec"] + "/nanopolish_dependencies_installed.success")
```
Then Snakemake will take care of touching the output file if the rule runs successfully.

Reply:
Removed if statement and changed output complete file as suggested